include/utils/filo.h

```diff
@@ -0,0 +1,120 @@
+#ifndef _UTILS_FILO_H_
+#define _UTILS_FILO_H_
+
+/*
+ * Copyright (c) 2020 Abhishek Arora <abhiarora@gotomain.io>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct {
+    void *buffer;
+    unsigned top;
+    unsigned elem_size;
+    unsigned max_size;
+} filo_t;
+
+#define FILO_DEF(type, name, sz) \
+    type name_##buffer[sz]; \
+    filo_t name = { .buffer = name ## _buffer, \
+                     .top = 0, \
+                     .elem_size = sizeof(type),\
+                     .max_size = sz \
+                    }; \
+
+#define FILO_GLOB_DEF(type, name, sz) \
```

Comment:
Glob here look odd as sounds like shell glob than implying global. If if you don't have a local varant of this macro, can we just call it `FILO_DEFINE()`?

Also, `FILO_EXTERN()` might be needed if someone wants to use it across files.

Reply:
Also, by convention the argument order has to be name, type, size.

Reply:
I have used the order `name, type and size` because it matches with C declaration of variables. For example, `int name[size];`

Reply:
When you have other macro helpers for the filo, they will all refer to the name param so keeping it as the first argument will help users recall calling order.

I know it doens't make sense in today's IDE era, but to a vim-er it helps when they can type out the params without looking it up.

Reply:
In my latest push, it was implemented the way you suggested.